src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+    const gl = this.gl;
+
+    this._updateSize();
+
+    const prevBoundTexture = gl.getParameter(gl.TEXTURE_BINDING_2D);
+    const prevBoundFramebuffer = gl.getParameter(gl.FRAMEBUFFER_BINDING);
+
+    const colorTexture = gl.createTexture();
+    if (!colorTexture) {
+      throw new Error('Unable to create color texture');
+    }
+    gl.bindTexture(gl.TEXTURE_2D, colorTexture);
+    const colorFormat = this._glColorFormat();
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      colorFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      colorFormat.format,
+      colorFormat.type,
+      null
+    );
+
+    // Create the depth texture
+    const depthTexture = gl.createTexture();
+    if (!depthTexture) {
+      throw new Error('Unable to create depth texture');
+    }
+    const depthFormat = this._glDepthFormat();
+    gl.bindTexture(gl.TEXTURE_2D, depthTexture);
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      depthFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      depthFormat.format,
+      depthFormat.type,
+      null
+    );
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, this.framebuffer);
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.COLOR_ATTACHMENT0,
+      gl.TEXTURE_2D,
+      colorTexture,
+      0
+    );
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.DEPTH_ATTACHMENT,
+      gl.TEXTURE_2D,
+      depthTexture,
+      0
+    );
+
+    // Create separate framebuffer for antialiasing
+    if (this.antialias) {
+      this.colorRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.colorRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
+        colorFormat.internalFormat,
+        this.width * this.density,
+        this.height * this.density
+      );
+
+      this.depthRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.depthRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
+        depthFormat.internalFormat,
+        this.width * this.density,
+        this.height * this.density
+      );
+
+      gl.bindFramebuffer(gl.FRAMEBUFFER, this.aaFramebuffer);
+      gl.framebufferRenderbuffer(
+        gl.FRAMEBUFFER,
+        gl.COLOR_ATTACHMENT0,
+        gl.RENDERBUFFER,
+        this.colorRenderbuffer
+      );
+      gl.framebufferRenderbuffer(
+        gl.FRAMEBUFFER,
+        gl.DEPTH_ATTACHMENT,
+        gl.RENDERBUFFER,
+        this.depthRenderbuffer
+      );
+    }
+
+    this.depthTexture = depthTexture;
+    this.colorTexture = colorTexture;
+
+    this.depth = new FramebufferTexture(this, 'depthTexture');
+    const depthP5Texture = new p5.Texture(
+      this.target._renderer,
+      this.depth,
+      {
```

Comment:
How come these are set to nearest?

Reply:
I'm assuming people probably won't want to draw the depth texture directly (some browsers treat it as only having a red channel while others mirror the depth across R, G, and B) but will be using it as an input in shaders for e.g. screen space ambient occlusion or shadow maps. I've mostly seen people [use NEAREST for ambient occlusion](https://learnopengl.com/Advanced-Lighting/SSAO) and apparently [the interpolation causes some unexpected edge cases in shadow maps](https://gamedev.stackexchange.com/questions/187280/is-linear-filtering-possible-on-depth-textures-in-opengl) so it seems like the safest option is to use NEAREST and let the programmer do some smoothing if they want it.